packages/beacon-node/src/chain/chain.ts

```diff
@@ -505,6 +506,36 @@ export class BeaconChain implements IBeaconChain {
     }
   }
 
+  prunePastInvalidSszObjects(): void {
+    const invalidObjectsDir = this.persistInvalidSszObjectsDir;
+    const invalidObjectsRetentionDays = this.opts.persistInvalidSszObjectsRetention ?? 7;
+    this.logger.debug("Pruning invalid ssz objects", {invalidObjectsDir, invalidObjectsRetentionDays});
+
+    const retentionDays = invalidObjectsRetentionDays + 1;
+    const DAYS_TO_MS = 24 * 60 * 60 * 1000;
+    const today = new Date();
+    const retentionDateInMS = today.getTime() - retentionDays * DAYS_TO_MS;
+    const basePath = invalidObjectsDir;
+
+    const datesPastRetention = (date: string): boolean => new Date(date).getTime() < retentionDateInMS;
+    fs.readdir(basePath, (err, list) => {
+      if (err) {
+        this.logger.debug("Unable to delete invalid ssz objects", {err: err.message});
+      } else {
+        for (const date of list.filter(datesPastRetention)) {
+          const dirpath = path.join(basePath, date);
+          fs.rm(dirpath, {recursive: true}, (err) => {
+            if (err) {
+              this.logger.debug("Failed to remove invalid ssz object", {dir: dirpath, err: err.message});
+            } else {
+              this.logger.debug("Removed invalid ssz object", {path: dirpath});
+            }
+          });
+        }
+      }
+    });
```

Comment:
We don't use the fs callback based API in Lodestar, just use the promise based and put a try catch at the top